lib/logstash/inputs/gelf.rb

```diff
@@ -96,7 +96,21 @@ def udp_listener(output_queue)
       # Gelfd parser outputs null if it received and cached a non-final chunk
       next if data.nil?
 
-      event = LogStash::Event.new(LogStash::Json.load(data))
+      begin
+        event = LogStash::Event.new(LogStash::Json.load(data))
+      rescue LogStash::Json::ParserError => e
+        @logger.info("JSON parse failure. Falling back to plain-text", :error => e, :data => data)
+        event = LogStash::Event.new("message" => data, "tags" => ["_jsonparsefailure"])
+      rescue StandardError => e
```

Comment:
I do not like this approach here. trying to protect logstash by rescuing `StandardError` is not helping solve the root problem at the right place. This code I believe has been taken from the json codec. I would prefer we use the json_lines codec pattern instead.

Currently logstash is a lot more resilient to crashes caused by plugin exceptions than it was prior to 2.x and we have a few in progress initiatives to improve it further.